src/core/internal/cppdemangle.d

```diff
@@ -0,0 +1,196 @@
+/**
+* This module provide the demangle for converts mangled C++ symbols
+*
+* Copyright: Copyright © 2019, The D Language Foundation
+* License: <a href="http://www.boost.org/LICENSE_1_0.txt">Boost License 1.0</a>.
+* Authors: Ernesto Castellotti
+* Source: $(DRUNTIMESRC core/internal/_cppdemangle.d)
+*/
+
+/**
+* Demangles C++ mangled names. If it is not a C++ mangled name, it returns its argument name.
+*
+ * Params:
+ *  buf = The string to demangle.
+ *
+ * Returns:
+ *  The demangled name or the original string if the name is not a mangled C++
+ *  name.
+ * The returns is always allocated with malloc, use free to deallocate the memory.
+ */
+@nogc @trusted nothrow char[] cppdemangle(const(char)[] buf)
+{
+    char[] fakeDst = null;
+    return cppdemangle(buf, fakeDst);
+}
+
+unittest
+{
+    version (Posix)
+    {
+        import std.stdio;
+        import core.stdc.stdlib : free;
+        auto demangledName = cppdemangle("_Znwm");
+        assert(demangledName == "operator new(unsigned long)");
+        free(demangledName.ptr);
+
+        extern(C++)  class Test
+        {
+            int var;
+        }
+
+        demangledName = cppdemangle(Test.var.mangleof);
+        writeln(demangledName);
+        assert(demangledName == "__unittest_L27_C1::Test::var");
```

Comment:
The `unittest` "function" is a D symbol. It would be much better to have this class defined outside of it.
But anyhow, this unittest depends on the C++ standard library to run, and I don't know if that's acceptable.